src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -108,14 +105,11 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundConstantPatte
         {
             if (guarded.Default != null)
             {
-                if (guarded.Default.MatchIsComplete)
-                {
-                    return null;
-                }
+                Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error
             }
             else
             {
-                guarded.Default = new DecisionTree.ByValue(guarded.Expression, guarded.Type, null);
+                guarded.Default = new DecisionTree.ByType(guarded.Expression, guarded.Type, null);
```

Comment:
>guarded.Default = new DecisionTree.ByType(guarded.Expression, guarded.Type, null); [](start = 16, length = 82)

Consider adding a comment why we should do this. Why change ```Default``` at all and this way in particular.